src/Lykke.Service.History/Workflow/Projections/TransactionHashProjection.cs

```diff
@@ -87,5 +87,33 @@ public async Task<CommandHandlingResult> Handle(CashoutCompletedEvent @event)
 
             return CommandHandlingResult.Ok();
         }
+
+        /// <summary>
+        ///     BIL batched cashout event
+        /// </summary>
+        /// <param name="event"></param>
+        /// <returns></returns>
+        public async Task<CommandHandlingResult> Handle(
+            Job.BlockchainCashoutProcessor.Contract.Events.CashoutsBatchCompletedEvent @event)
+        {
+            if (@event.Cashouts == null || @event.Cashouts.Length == 0)
+            {
+                _logger.Warning($"BIL batched cashout event, BatchId {@event.BatchId}", context: @event);
+
+                return CommandHandlingResult.Ok();
+            }
+
+            foreach (var cashout in @event.Cashouts)
+            {
+                if (!await _historyRecordsRepository.UpdateBlockchainHashAsync(cashout.OperationId, @event.TransactionHash))
+                    _logger.Warning($"Transaction hash was not set, BIL cashout", context: new
```

Comment:
This means that some data is in not consistent state, we should throw an exception to retry, not just log a warning